cmd/zpool/zpool_main.c

```diff
@@ -410,8 +421,12 @@ zpool_collect_leaves(zpool_handle_t *zhp, nvlist_t *nvroot, nvlist_t *res)
 	    &child, &children);
 
 	if (children == 0) {
-		char *path = zpool_vdev_name(g_zfs, zhp, nvroot, B_FALSE);
-		fnvlist_add_boolean(res, path);
+		char *path = zpool_vdev_name(g_zfs, zhp, nvroot,
+		    VDEV_NAME_PATH);
+
+		if (strcmp(path, VDEV_TYPE_INDIRECT) != 0)
```

Comment:
This was added mainly for cosmetic reasons in order to avoid passing an indirect vdev guid to the kernel.  This is handled properly on the kernel side by returning an error, but it results in an ugly warning whenever issuing a `zpool trim <pool>` for a pool containing an indirect vdev.  Detecting it here where we know the type was the cleanest way I found to handle this.